tests/units/utilities/test_dataframe_functions.py

```diff
@@ -1089,3 +1089,47 @@ def test_dict_input(self):
         self.assertEqual(list(extended_dataframe["id"]),
                          ["0,1", "0,1", "1,1", "1,1", "0,2", "0,2", "1,2", "1,2", "2,2"])
         assert_series_equal(dataframe["value"], extended_dataframe["value"])
+
+
+class PivotListTestCase(TestCase):
+    def test_empty_list(self):
+        return_df = dataframe_functions.pivot_list([])
+
+        self.assertEqual(len(return_df), 0)
+        self.assertEqual(len(return_df.index), 0)
+        self.assertEqual(len(return_df.columns), 0)
+
+    def test_different_input(self):
+        input_list = [
+            ("a", "b", 1),
+            ("a", "c", 2),
+            ("A", "b", 3),
+            ("A", "c", 4),
+            ("X", "Y", 5),
+        ]
+        return_df = dataframe_functions.pivot_list(input_list)
+
+        self.assertEqual(len(return_df), 3)
+        self.assertEqual(set(return_df.index), {"a", "A", "X"})
+        self.assertEqual(set(return_df.columns), {"b", "c", "Y"})
+
+        self.assertEqual(return_df.loc["a", "b"], 1)
+        self.assertEqual(return_df.loc["a", "c"], 2)
+        self.assertEqual(return_df.loc["A", "b"], 3)
+        self.assertEqual(return_df.loc["A", "c"], 4)
+        self.assertEqual(return_df.loc["X", "Y"], 5)
+
+    def test_long_input(self):
+        input_list = []
+        for i in range(100):
```

Comment:
You could use https://docs.python.org/3/library/itertools.html#itertools.product here to get rid of the loops

Reply:
That is true - but I wanted to keep the unittest as "simple" as possible :-) 
I am actually not sure which possibility is easier...